File: DSA/Pragya_Agarwal/Day16_Question2.py
class Node:
    def __init__(self,val):
        self.data = val
        self.left = None
        self.right = None

def InOrder(node):
    if node == None:
        return 
    
    InOrder(node.left)
    print(node.data,end=" ")
    InOrder(node.right)

if __name__ == '__main__':
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)

print("Inorder Traversal:")
InOrder(root)
    

File: DSA/RAJAS-BAGGA/Day_16_Question_2.java

import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

public class Day_16_Question_2 {
    static List<Integer> tree = new ArrayList<>();
    public static List<Integer> inOrder(TreeNode root) {
        traverse(root);
        return tree;
    }
    public static void traverse(TreeNode root) {
        
        if (root == null)
            return;
        
        traverse(root.left);
        tree.add(root.val);
        traverse(root.right);
    }
    
    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        TreeNode n1 = new TreeNode(2);
        TreeNode n2 = new TreeNode(3);
        TreeNode n3 = new TreeNode(4);
        TreeNode n4 = new TreeNode(5);
        
        root.left = n1;
        root.right = n2;
        n1.left = n3;
        n1.right = n4;
        System.out.println(inOrder(root));
    }
}
